internal/apex/wg-iface.go

```diff
@@ -0,0 +1,195 @@
+		wgIP := getIPv4Iface(wgIface)
+		_, err = RunCommand("ip", "address", "del", wgIP.To4().String(), "dev", wgIface)
+		if err != nil {
+			logger.Errorf("failed to assign an address to the local linux interface: %v\n", err)
+		}
+		_, err = RunCommand("ip", "address", "add", ax.wgLocalAddress, "dev", wgIface)
+		if err != nil {
+			logger.Errorf("failed to assign an address to the local linux interface: %v\n", err)
+		}
+	}
+	// bring the wg0 interface up
+	_, err = RunCommand("ip", "link", "set", wgIface, "up")
+	if err != nil {
+		logger.Errorf("failed to bring up the wg interface: %v\n", err)
+	}
+}
+
+func setupWindowsIface(logger *zap.SugaredLogger, localAddress, privateKey string) error {
+	if err := buildWindowsWireguardIfaceConf(privateKey, localAddress); err != nil {
+		return fmt.Errorf("failed to create the windows wireguard wg0 interface file %v", err)
+	}
+
+	var wgOut string
+	var err error
+	if ifaceExists(logger, wgIface) {
+		wgOut, err = RunCommand("wireguard.exe", "/uninstalltunnelservice", wgIface)
+		if err != nil {
+			logger.Debugf("failed to down the wireguard interface (this is generally ok): %v\n", err)
+		}
+		time.Sleep(time.Second * 1)
+	}
+	logger.Debugf("stopped windows tunnel svc:%v\n", wgOut)
+	// sleep for one second to give the wg async exe time to tear down any existing wg0 configuration
+	wgOut, err = RunCommand("wireguard.exe", "/installtunnelservice", windowsWgConfigFile)
+	if err != nil {
+		return fmt.Errorf("failed to start the wireguard interface: %v\n", err)
+	}
+	logger.Debugf("started windows tunnel svc: %v\n", wgOut)
+	time.Sleep(time.Second * 1)
+
+	return nil
+}
+
+// deleteDarwinIface delete the darwin userspace wireguard interface
+func deleteDarwinIface(logger *zap.SugaredLogger) {
+	tunSock := fmt.Sprintf("/var/run/wireguard/%s.sock", darwinIface)
+	_, err := RunCommand("rm", "-f", tunSock)
+	if err != nil {
+		logger.Debugf("failed to delete darwin interface: %v", err)
+	}
+	// /var/run/wireguard/wg0.name doesnt currently exist since utun8 isnt mapped to wg0 (fails silently)
+	wgName := fmt.Sprintf("/var/run/wireguard/%s.name", wgIface)
+	_, err = RunCommand("rm", "-f", wgName)
+	if err != nil {
+		logger.Debugf("failed to delete darwin interface: %v", err)
+	}
+}
+
+// ifaceExists returns true if the input matches a net interface
+func ifaceExists(logger *zap.SugaredLogger, iface string) bool {
+	_, err := net.InterfaceByName(iface)
+	if err != nil {
+		logger.Debugf("existing link not found: %s", iface)
+		return false
+	}
+
+	return true
+}
+
+// getIPv4Iface get the IP of the specified net interface
+func getIPv4Iface(ifname string) net.IP {
+	interfaces, _ := net.Interfaces()
+	for _, inter := range interfaces {
+		if inter.Name == ifname {
+			if addrs, err := inter.Addrs(); err == nil {
+				for _, addr := range addrs {
+					switch ip := addr.(type) {
+					case *net.IPNet:
+						if ip.IP.DefaultMask() != nil {
+							return ip.IP
+						}
+					}
+				}
+			}
+		}
+	}
```

Comment:
Similar to an earlier comment, my style preference would be to cut down on the nesting depth like this, but up to you if you want to take it or not.

```suggestion
for _, inter := range interfaces {
	if inter.Name != ifname {
		continue
	}
	if addrs, err := inter.Addrs(); err != nil {
		continue
	}
	for _, addr := range addrs {
		switch ip := addr.(type) {
		case *net.IPNet:
			if ip.IP.DefaultMask() != nil {
				return ip.IP
			}
		}
	}
}
```